cuqi/distribution/_cauchy.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+import scipy.stats as sps
+from cuqi.geometry import _get_identity_geometries
+from cuqi.utilities import force_ndarray
+from cuqi.distribution import Distribution
+
+class Cauchy(Distribution):
+    """ Multivariate Cauchy distribution of independent random variables.
+     
+    Each is distributed according to the PDF function
+
+    .. math::
+
+        \\frac{1}{\\pi\\gamma(1+\\frac{(x-\\mu)^2}{\\gamma^2})}
+
+    where :math:`\\mu` is the location parameter and :math:`\\gamma` is the scale parameter.
+
+    Parameters
+    ----------
+    location: float or array_like
+        Location parameter
+
+    scale: float or array_like
+        Scale parameter
+
+    Example
+    -------
+    .. code-block:: python
+
+        # % Generate a Cauchy distribution
+        import cuqi
+        location = 0
+        scale  = 1
+        x = cuqi.distribution.Cauchy(location, scale)
+        x.logpdf(0) # -1.1447298858494002
+
+    """
+
+    def __init__(self, location=None, scale=None, is_symmetric=True, **kwargs):
+        super().__init__(is_symmetric=is_symmetric, **kwargs)
+        self.location = location
+        self.scale = scale
+
+    @property
+    def location(self):
+        """ Location parameter """
+        return self._location
+    
+    @location.setter
+    def location(self, value):
+        self._location = force_ndarray(value, flatten=True)
+
+    @property
+    def scale(self):
+        """ Scale parameter """
+        return self._scale
+    
+    @scale.setter
+    def scale(self, value):
+        self._scale = force_ndarray(value, flatten=True)
+
+    def _is_out_of_bounds(self, x):
+        """ Check if x is out of bounds """
+        return np.any(self.scale <= 0)
```

Comment:
Given that the Cauchy distribution PDF is defined on (-inf, inf) and that scale should be greater than zero, I am wondering if we need this function. The function does not actually use the value of x. Could an alternative approach be to test inside the scale setter:
```python
if np.any(value <= 0):
    #raise some error
```

Reply:
We do not want to raise an error since sampling algorithms may propose these kinds of values and we want the logpdf to return a valid value, but impossible.